tests/pester-tests/features/CredentialProvider.Tests.ps1

```diff
@@ -0,0 +1,62 @@
+﻿Describe 'Ensuring credentials do not bleed from configured sources' -Tag CredentialProvider -ForEach @(
+    @{
+        Command = 'info'
+        ExitCode = 0
+}
+    @{
+        Command = 'install'
+        ExitCode = 1
+}
+    @{
+        Command = 'outdated'
+        ExitCode = 0
+}
+    @{
+        Command = 'search'
+        ExitCode = 1
+}
+    @{
+        Command = 'upgrade'
+        ExitCode = 1
+}
+    @{
+        Command = 'download'
+        ExitCode = 1
+}
+) {
+    BeforeDiscovery {
+        $HasLicensedExtension = Test-PackageIsEqualOrHigher -PackageName 'chocolatey.extension' -Version '6.0.0'
+    }
+
+    BeforeAll {
+        Initialize-ChocolateyTestInstall
+        Disable-ChocolateySource -All
+        Enable-ChocolateySource -Name 'hermes'
+        $SetupSource = Get-ChocolateySource -Name 'hermes-setup'
+        Remove-Item download -force -recurse
+    }
+
+    # Skip the download command if chocolatey.extension is not installed.
+    Context 'Command (<Command>)' -Skip:($Command -eq 'download' -and -not $HasLicensedExtension) {
+        BeforeAll {
+            # Picked a package that is on `hermes-setup` but not on `hermes`.

```

Comment:
How are we going to enforce this going forward?

Reply:
I've updated this comment. The package used doesn't really matter, but it is best if we use one that might be found, so I used `chocolatey`.

Reply:
I've re-run the tests. The package needs to be one that is not already installed (otherwise `choco install won't try to install it), but may be found on the repository. As such, I've switched it back to `chocolatey-compatibility.extension` so that the tests all pass.